### PR TITLE
Add rebase workflow

### PR DESCRIPTION
**Pull request by timflannagan**
Introduce a rebase workflow responsible for automatically rebasing open
pull requests when changes merge to the default branch (main). Pull
requests opened from forks require pull requests authors to enable edits
from maintainers in order to successfully rebase PRs.

Signed-off-by: timflannagan <timflannagan@gmail.com>

### CODE REVIEW

File: .github/workflows/rebase.yaml
Comment (timflannagan) [inline .github/workflows/rebase.yaml:12]: For more context, it looks like you can achieve some of this using native branch protection policies on the default branch (main in this case) and we should enable that, but it's still a manual effort in terms of rebasing content. Other tooling around rebasing PRs when changes land in master have different stories specifically around forks opened PRs varying from it's not supported, to you need to have a custom PAT token configured for the repository, vs. this one which just requires allowing maintainers to edit the pull request content.

Comment (timflannagan) [reply .github/workflows/rebase.yaml:12]: Link to the allowing 'edits from maintainers' documentation: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork.

Comment (tylerslaton) [reply .github/workflows/rebase.yaml:12]: Is this the setting you're talking about in the first comment?

Screen Shot 2021-11-11 at 3 49 48 PM

Comment (timflannagan) [reply .github/workflows/rebase.yaml:12]: Yeah, those are the checks. You'll need to manually define the required checks when I poking around those options earlier today, but I think that should give us the requisite coverage for 1.) automatic rebasing of PRs once changes land in the main branch and 2.) decrease the chance of regressing main for open PRs.

Comment (tylerslaton) [reply .github/workflows/rebase.yaml:12]: Got it. I believe that this should work once we add this rebase check you're merging so this LGTM.

**Review by tylerslaton — APPROVED** [review]
LGTM! Thanks for doing this.

**Pull request merged**
